### PR TITLE
Create automatic test for logic components and calculator rendered buttons

### PR DESCRIPTION
**Pull request by Sirvelasque**
## Set up and create tests

:white_check_mark: *Set up React testing library* 
:white_check_mark: *Create test for calculate component*
:white_check_mark: *Create test for operate component*
:white_check_mark: *Use React Testing Library to simulate user interaction.*
:white_check_mark: *Use Jest snapshots to test the components.*

### CODE REVIEW

File: src/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Snapshot test test App should render and match the Snapshot 1`] = `null`;
```
Comment (ahmetbozaci) [inline src/__snapshots__/index.test.js.snap:3]: - [ ] Kindly add more snapshot tests for other components(header, button, app) [Requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md#:~:text=Use%20Jest%20snapshots%20to%20test%20the%20components.)

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @Sirvelasque @BregornOriginal  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- Calculator working well. ✔️
- All tests are passed with no error.✔️
___


### ♻️ Request changes ♻️

- [x] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [ ] It has one warning. Easy to fix. You can fix it.
image

@BregornOriginal 
- Your commit messages should always start with capital letters and they should be an imperative tense. I see some of them are not like that. Instead of writing `Added style or Created the file` you should write `Add style or Create the file`
- ***I write it as optional. But this is a [general requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=The%20commit%20message%20should%20be%20written%20in%20the%20imperative%20tense%20and%20capitalized) for every project. So you should pay attention to this in your next projects.***

**OPTIONAL**

- You can change them by using `git-rebase`. If you want to know how to change ı leave a few resources.
- [Video - How to change commit messages](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6)
- [Article - How to create good commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: src/index.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const Container = () => {
+    <BrowserRouter>
+      <App />
+    </BrowserRouter>;
+  };
+  it('test App should render and match the Snapshot', () => {
+    const tree = renderer.create(<Container />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
[the remark this answers is not in the document]
Comment (BregornOriginal) [reply src/index.test.js]: Thank you for the suggestions @ahmetbozaci

**Review by Deepakdanger — APPROVED** [review]
Hi @Team, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Kindly include all the author's detail in the Readme file.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**